tests/v2/validation/configmaps/generated_fields_test.go

```diff
@@ -0,0 +1,109 @@
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+	"testing"
+)
+
+const (
+	APIVersion         = "v1"
+	Kind               = "ConfigMap"
+	configMapNamespace = "default"
+)
+
+type ConfigMapTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	steveClient        *steveV1.Client
+	session            *session.Session
+	cluster            *management.Cluster
+	nameSpacedV1Client *steveV1.NamespacedSteveClient
+	configMapPayload   *cm.SteveConfigMap
+}
+
+func (c *ConfigMapTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ConfigMapTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+	client, err := rancher.NewClient("", c.session)
+	require.NoError(c.T(), err)
+	c.client = client
+	c.steveClient = client.Steve
+}
+
+func (c *ConfigMapTestSuite) TestSteveGeneratedFields() {
+
+	steveClient := c.client.Steve
+	configMapName := namegenerator.AppendRandomString("test-configmap")
+
+	v1ConfigMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: configMapNamespace,
+		},
+		Data: map[string]string{
+			"env": "qa",
+		},
+	}
+
+	configmapSteveObject, err := steveClient.SteveType(cm.ConfigMapSteveType).Create(v1ConfigMap)
+	require.NoError(c.T(), err)
+
+	c.configMapPayload = &cm.SteveConfigMap{}
+	setPayload(configmapSteveObject, c.configMapPayload)
+
+	headers, _, err := steveClient.SteveType("configmaps").NamespacedSteveClient(configMapNamespace).PerformPutCaptureHeaders(c.client.RancherConfig.Host, c.client.RancherConfig.AdminToken, c.configMapPayload.Name, c.configMapPayload)
+	require.NoError(c.T(), err)
+
+	warnings, ok := headers["Warning"]
+	var failWarnings []string
+	if ok {
+		for _, warning := range warnings {
+			if strings.HasPrefix(warning, "299") {
+				logrus.Printf("Warning header found: %s", warning)
+				failWarnings = append(failWarnings, warning)
+			}
+		}
+	}
+
+	if len(failWarnings) > 0 {
+		require.Fail(c.T(), fmt.Sprintf("Test failed due to warnings: \n%s", strings.Join(failWarnings, "\n")))
+	}
+}
+
+func TestConfigMapTestSuite(t *testing.T) {
+	suite.Run(t, new(ConfigMapTestSuite))
+}
+
+func setPayload(apiObject *steveV1.SteveAPIObject, payload *cm.SteveConfigMap) {
+	payload.APIVersion = APIVersion
+	payload.Kind = Kind
+	payload.Name = apiObject.Name
+	payload.Namespace = configMapNamespace
+	payload.ResourceVersion = apiObject.ResourceVersion
+	payload.UID = apiObject.UID
+	payload.State = &steveV1.State{}
+	payload.Relationships = &[]steveV1.Relationship{}
+	payload.Fields = []interface{}{"foo", "bar"}
+	data, ok := apiObject.JSONResp["data"].(map[string]interface{})
+	if !ok {
+		logrus.Println("Type assertion failed")
+		return
+	}
```

Comment:
I think we can expand this check a bit; If the key is not found, we can return an error here. And use require/assert in the test to make sure there's no error. That way we can assert or exit the test if this is a necessary step with `require`

Reply:
Updated ✅